homeassistant/components/sensor/recorder.py

```diff
@@ -375,21 +380,20 @@ def compile_statistics(  # noqa: C901
                     )
                 continue
 
-        result[entity_id] = {}
-
         # Set meta data
-        result[entity_id]["meta"] = {
+        meta: StatisticMetaData = {
+            "statistic_id": entity_id,
```

Comment:
Are we going to allow integrations to define their own statistic IDs when importing? I guess we can, but we should enforce it starts with `<domain>-`